app/views/collection_drafts/forms/fields/_use_constraints_fields.html.erb

```diff
@@ -1,46 +1,63 @@
   ) %>
-  <%= mmt_help_icon(
-    title: 'Description',
-    help: 'definitions/UseConstraintsType/properties/Description'
-  ) %>
   <%= render partial: 'collection_drafts/forms/type', locals: {
     type: 'use_constraints_description',
     values: use_constraints['Description'] || {},
     name_prefix: "#{name_prefix}|description|_",
     multiple: false
   } %>
 
-  <!-- License URL -->
-  <%= mmt_label(
-    name: 'license_url',
-    title: 'License Url',
-    prefix: name_prefix
-  ) %>
-  <%= mmt_help_icon(
-    title: 'License Url',
-    help: 'definitions/UseConstraintsType/properties/LicenseUrl'
-  ) %>
-  <%= render partial: 'collection_drafts/forms/type', locals: {
-    type: 'online_resource',
-    values: use_constraints['LicenseUrl'] || {},
-    name_prefix: "#{name_prefix}|license_url|_",
-    multiple: false
-  } %>
+  <div class="row sub-fields">
+    <!-- Constraint type selector -->
+    <%= mmt_label(
+      name: 'use_constraint_type',
+      title: 'Constraint Type',
+      prefix: name_prefix,
+      always_required: false
+    ) %>
+    <div class='use_constraint_type_group row sub-fields'>
+      <p class="radio-group"><%= radio_button_tag(name_to_param("#{name_prefix}"), 'LicenseUrl', license_url_checked, class: 'license-available-select', name: nil, data: { level: remove_pipes(name_prefix) }) %> <label for="<%= remove_pipes(name_prefix) %>LicenseUrl">License Url</label></p>
+      <p class="radio-group"><%= radio_button_tag(name_to_param("#{name_prefix}"), 'LicenseText', license_text_checked, class: 'license-available-select', name: nil, data: { level: remove_pipes(name_prefix) }) %><label for="<%= remove_pipes(name_prefix) %>LicenseText">License Text</label></p>
+    </div>
+  </div>
+
+    <!-- License Url -->
+    <div class="license-url-fields available row sub-fields" style="<%= 'display: none' unless license_url_checked %>">
+      <%= mmt_label(
+        name: 'license_url',
+        title: 'License Url',
+        prefix: name_prefix
+      ) %>
+      <%= mmt_help_icon(
+        title: 'License Url',
+        help: 'definitions/UseConstraintsType/properties/LicenseUrl'
+      ) %>
+      <%= render partial: 'collection_drafts/forms/type', locals: {
+        type: 'online_resource',
+        values: use_constraints['LicenseUrl'] || {},
+        name_prefix: "#{name_prefix}|license_url|_",
+        multiple: false
+      } %>
+    </div>
 
-  <!-- License Text -->
-  <%= mmt_text_area(
-    name: 'license_text',
-    title: 'License Text',
-    prefix: name_prefix,
-    value: use_constraints['LicenseText'],
-    help: 'definitions/UseConstraintsType/properties/LicenseText',
-    validate: true,
-    required: true
-   ) %>
-</div>
+    <!-- License Text -->
+    <div class="license-text-fields not-available row sub-fields" style="<%= 'display: none' unless license_text_checked%>">
+      <%= mmt_text_area(
+        name: 'license_text',
+        title: 'License Text',
+        prefix: name_prefix,
```

Comment:
if you change this to `"#{name_prefix}|license_text|_"`, it should alter the data-level enough so it acts similar to the License Url grouping for the one field. if there is data in the field the required icon will appear, if data is removed, the icon goes away.

Reply:
After further testing, this change does work for the required icons issue. However, it also creates a problem with the name of the field (and therefore saving and displaying the value). I have a potential solution, but I think it would be best to discuss the best course of action here on a screen share so let me know when you have some time for that.

Reply:
I cleaned up some of the html and made this change.